Tests/MoyaTests/MoyaProviderIntegrationTests.swift

```diff
@@ -119,18 +119,19 @@ final class MoyaProviderIntegrationTests: QuickSpec {
                     }
 
                     it("uses a background queue") {
-                        var isMainThread: Bool?
+
+                        let isMainThread = Atomic<Bool?>(wrappedValue: nil)
                         let callbackQueue = DispatchQueue(label: "background_queue", attributes: .concurrent)
                         let target: GitHub = .zen
 
                         waitUntil { done in
                             provider.request(target, callbackQueue: callbackQueue) { _ in
-                                isMainThread = Thread.isMainThread
+                                isMainThread.wrappedValue = Thread.isMainThread
```

Comment:
I know that this was there before, but this is probably a bad check - main thread can be reused in a background queue, so we probably should check for a specific queue instead (similar to the labels we used before)

Reply:
That makes sense, but not sure we should address this here, I think we could put a FIXME comment and address it in a separated PR :)

Reply:
Never mind, It's too simple ... I just changed :)

Reply:
thank you!